core/src/main/java/pmstation/core/plantower/ParticulateMatterSample.java

```diff
@@ -30,18 +33,30 @@ public ParticulateMatterSample(int pm1_0, int pm2_5, int pm10) {
     public ParticulateMatterSample(int pm1_0, int pm2_5, int pm10,
             int hcho, int humidity, int temperature,
             byte modelVersion, byte errCode) {
-
+        pmMap = new HashMap<>();
         date = new Date();
         this.hcho = hcho; // ug/m^3
                 // see levels in mg/m^3:
                 // http://archiwum.ciop.pl/13999 (quite old, 1995),
                 // or here in ug/m^3:
                 // https://www.canada.ca/content/dam/canada/health-canada/migration/healthy-canadians/publications/healthy-living-vie-saine/formaldehyde/alt/formaldehyde-eng.pdf (2006)
+        addFieldToMap("hcho", Map.of("value", hcho, "unit", "µg/m³"));
+
         this.humidity = humidity >= 0 ? (double) humidity / 10.0 : -1; // %
+        addFieldToMap("hcho", Map.of("value", humidity, "unit", "%"));
```

Comment:
that's why, pls create enum with those keys and their units. You can even define enum with "getter" - then it may 'automatically' call right method to get the value (for example using https://docs.oracle.com/en/java/javase/15/docs/api/java.base/java/util/function/Function.html or Supplier?). See https://github.com/mucommander/mucommander/blob/80d237cde86ea7f5e956dbbc4603c14a937aa019/mucommander-viewer-text/src/main/java/com/mucommander/viewer/text/TextViewerPreferences.java#L122 for some inspiration.

Reply:
1st we need to resurect Android version before we will modify or add another wrapper.

Reply:
naah... just add this enum inside this ParticulateMatterSample.